### DISCUSSION TITLE
Carlos Perez: strawberry takes "turns"

### DISCUSSION THREAD

**Discussion by daveshap** [daveshap/Raspberry#31] [Research]
https://x.com/IntuitMachine/status/1835256547179413672

![image](https://github.com/user-attachments/assets/a50a4d72-e7cc-432b-9136-fc582c8d971a)

**Comment by russellballestrini** [comment]
The reasoning tokens are discarded from context for the end user to prevent poisoning the context but the tokens that make the chain are certainly saved and used to train.

**Comment by russellballestrini** [comment]
We seem to have counter claims of multi turn or single shot.

**Comment by Neoathenian** [comment]
It doesn´t seem possible to me for it to be a single shot, how would they limit the amount of compute if so?
I feel like the innovation is probably a model that reasons better, and then the normal CoT with that model.

Regarding this post, the image looks weird because it seems unlikely to me that the model would be able to catch itself in it´s errors if it only saw the ouptut and not it´s reasoning

**Reply by daveshap** [reply]
it's a multi-phase output. I already demonstrated how the training data is formed.

**Reply by Neoathenian** [reply]
If I remember correctly the training data you showed was basically including the reasoning steps in it too (so CoT).
The issue I have is that this wouldn´t allow you to say, have the LLM produce an output for 10 days straight to get a better response. (Like control if it´s 1 day or 10 days)

**Reply by daveshap** [reply]
That's not necessarily true. Noam Brown said they trained it to just keep producing tokens until it gets an answer. There's no evidence that the token window is large enough to generate for "10 days straight" and we're not trying to supersede SOTA. The longest generation I've seen from o1-preview is about 5 minutes.

**Reply by ElliottDyson** [reply]
> That's not necessarily true. Noam Brown said they trained it to just keep producing tokens until it gets an answer. There's no evidence that the token window is large enough to generate for "10 days straight" and we're not trying to supersede SOTA. The longest generation I've seen from o1-preview is about 5 minutes. 

You could also increase or decrease the likelihood for it to generate the token that shifts it out of chain-of-thought based on how long you want generations to occur. I imagine openAI must have done something like this in order to obtain their graphs on accuracy Vs compute time at inference via introducing a new token for this purpose.

**Reply by russellballestrini** [reply]
You would maybe want to signal it to finish its thought and end properly so the tags can be closed and output parsed (segregated) nicely from the solution or answer.

**Reply by ElliottDyson** [reply]
> You would maybe want to signal it to finish its thought and end properly so the tags can be closed and output parsed (segregated) nicely from the solution or answer.

Good idea, like a second token that we might choose to introduce that indicates to the model to finish its thoughts soon and end them with an COT-end token
